File: JAVA/1790. Check if One String Swap Can Make Strings Equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        char[] a=s1.toCharArray();
        char[] b=s2.toCharArray();
        int i,c=0;
        for(i=0;i<a.length;i++){
            if(a[i]!=b[i])
            c++;
        }
        if(c==0)
        return true;
        if(c==2){
            Arrays.sort(a);
            Arrays.sort(b);
            return (Arrays.equals(a,b));
        }
        return false;
    }
}

File: C++/1790. Check if One String Swap Can Make Strings Equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        int i,c=0;
        vector<char>a;
        for(i=0;i<s1.size();i++){
            if(s1[i]!=s2[i]){
                a.push_back(s1[i]);
                a.push_back(s2[i]);
                if(a.size()>4)
                return false;
            }
        }
        if(a.size()==0)
        return true;
        if(a.size()==2)
        return false;
        return(a[0]==a[3] && a[1]==a[2]);
    }
};